pymc/logprob/order.py

```diff
@@ -158,3 +183,54 @@ def max_logprob_discrete(op, values, base_rv, **kwargs):
 
     n = pt.prod(base_rv_shape)
     return logdiffexp(n * logcdf, n * logcdf_prev)
+
+
+class MeasurableSort(MeasurableOp, SortOp):
+    """A placeholder used to specify a log-likelihood for a sort sub-graph."""
+
+
+@_logprob.register(MeasurableSort)
+def sort_logprob(op, values, base_rv, axis, **kwargs):
+    r"""Compute the log-likelihood graph for the `Sort` operation."""
+    (value,) = values
+
+    logprob = _logprob_helper(base_rv, value).sum(axis=-1)
+
+    base_rv_shape = constant_fold(tuple(base_rv.shape), raise_not_constant=False)
+    n = pt.prod(base_rv_shape, axis=-1)
+    sorted_logp = pt.log(n) + logprob
```

Comment:
This doesn't seem to be the correct logp. I looked it up in the wikipedia page you linked [here](https://en.wikipedia.org/wiki/Order_statistic#Cumulative_distribution_function_of_order_statistics) and it's much more complicated than that.

To get it, we need the `base_rv` logcdf too:
<img width="531" alt="imagen" src="https://github.com/user-attachments/assets/e14fd4bd-afaf-4355-b8ff-1713b1b5aec7">

Reply:
So logprob would be something like

```python
r = pt.arange(n)
sorted_logprob = (
    gamma_ln(n + 1) - gamma_ln(r) - gamma_ln(n - r)
    + logprob
    + (r - 1) * logcdf
    + (n - r) * logcdf
)
```

Reply:
Using that same link, you could also register a logcdf expression using this:
<img width="383" alt="imagen" src="https://github.com/user-attachments/assets/4de2880c-2ad9-4e0e-b8f8-5529699cba22">

It would look something like
```python
r = pt.arange(n)
sorted_logcdf = logsumexp(
    gamma_ln(n + 1) - gamma_ln(r + 1) - gamma_ln(n - r)
    + r * logcdf
    + (n - r) * logcdf
)
```

Reply:
Oh no. I gave this some more thought and realized that the expressions I pasted above were for a single draw, not for the joint distribution. And the part that you posted about the joint says

<img width="615" alt="imagen" src="https://github.com/user-attachments/assets/ab477710-0c55-4855-959d-ed4bdfce2eff">

So your expression is almost correct, except for the log(n) instead of log(n!)

```
sorted_logp = pt.gamma_ln(n + 1) + logprob
```

Reply:
Yes, for order stats you need cdfs, you chan check the max logp (same file) uses it.

Reply:
fixed